### PR TITLE
Add a lambda to handle feast google purchases

### PR DESCRIPTION
**Pull request by MaelGNM**
## What does this change?
This provides the webhook for the store to send purchase informations to. The lambda doesn't do anything yet.

## How to test
Call the endpoint in CODE and it should return a 200 echoing the request body.
I tested with postman and it worked as intended.

### CODE REVIEW

File: cloudformation.yaml [hunk 1]
```diff
@@ -975,6 +1021,35 @@ Resources:
...
+          Value: /feast/google/pubsub
+        - Name: Stage
+          Value: !Ref Stage
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      EvaluationPeriods: 1
+      MetricName: 5XXError
+      Namespace: AWS/ApiGateway
+      Period: 300
+      Statistic: Sum
+      Threshold: 2
```
Comment (johnduffell) [inline cloudformation.yaml:1050]: if the description is indended as "A Http request.... (failed)" then it should be 1.
Also for "1" error, the alarm comes through up to 4 minutes quicker if it's a 1 minute period
```suggestion
      Period: 60
      Statistic: Sum
      Threshold: 1
```

File: typescript/src/feast/pubsub/google.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import {APIGatewayProxyEvent, APIGatewayProxyResult} from "aws-lambda";
+
+export async function handler(request: APIGatewayProxyEvent): Promise<APIGatewayProxyResult>  {
+    console.log(`${JSON.stringify(request)}`)
```
Comment (johnduffell) [inline typescript/src/feast/pubsub/google.ts:4]: hardly worth adding now, but it makes it slightly easier to review if there's a comment explaining the body
```suggestion
    // placeholder for new lambda
    console.log(`${JSON.stringify(request)}`)
```

File: cloudformation.yaml [hunk 3]
```diff
@@ -975,6 +1021,35 @@ Resources:
...
+        - Ref: AlarmTopic
+      AlarmName: !Sub mobile-purchases-${Stage}-feast-google-pubsub-check-errors
+      AlarmDescription: A HTTP request to the Feast Google pubsub endpoint resulted in a 5XX error.
+      Dimensions:
+        - Name: ApiName
+          Value: !Sub ${App}-${Stage}
+        - Name: Method
+          Value: POST
+        - Name: Resource
+          Value: /feast/google/pubsub
```
Comment (johnduffell) [inline cloudformation.yaml:1041]: alarms are notoriously hard to test - I am not sure if it's worth forcing a 5xx error and making sure a metric does end up in the right place with the right dimensions?

File: cloudformation.yaml [hunk 4]
```diff
@@ -305,6 +309,17 @@ Resources:
...
+            post:
+              x-amazon-apigateway-integration:
+                httpMethod: POST
+                type: aws_proxy
+                uri: !Sub arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${FeastGooglePubSubLambda.Arn}/invocations
+              consumes: [application/json]
+              produces: [application/json]
+              responses:
+                "200":
+                  "description": "200 response"
```
Comment (johnduffell) [inline cloudformation.yaml:322]:
```suggestion
                  "description": "we successfully processed an incoming ,,,,,something.....?"
```

**Review by johnduffell — APPROVED** [review]
looks good, just some small comments (I don't know this code base at all, so I was just revewing it to get an idea of it!)

File: cloudformation.yaml (hunk 1 above)
Comment (MaelGNM) [reply cloudformation.yaml:1050]: Yes, I'm wondering what we actually want from this alarm. We have the same alarm rule for all the pubsub endpoints, and we'd probable want to keep them the same whichever way we end up going.

File: typescript/src/feast/pubsub/google.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import {APIGatewayProxyEvent, APIGatewayProxyResult} from "aws-lambda";
```
Comment (tjmw) [inline typescript/src/feast/pubsub/google.ts:1]: Very minor, but is it worth being explicit that these are types?

```suggestion
import type {APIGatewayProxyEvent, APIGatewayProxyResult} from "aws-lambda";
```

**Review by tjmw — APPROVED** [review]
Nice!

File: cloudformation.yaml (hunk 4 above)
Comment (MaelGNM) [reply cloudformation.yaml:322]: Will address in a separate PR improving the other descriptions

File: cloudformation.yaml (hunk 1 above)
Comment (MaelGNM) [reply cloudformation.yaml:1050]: Will address in a separate PR, trying to improve the alarm for all 4 endpoints

File: typescript/src/feast/pubsub/google.ts (hunk 5 above)
Comment (MaelGNM) [reply typescript/src/feast/pubsub/google.ts:1]: Good point

File: cloudformation.yaml (hunk 3 above)
Comment (MaelGNM) [reply cloudformation.yaml:1041]: I think it's worth talking about alarms for this stack in general (especially since they're easy to miss)

**Pull request merged**